USER_GUIDE.md

```diff
@@ -1,6 +1,8 @@
 - [User Guide](#user-guide)
-    - [Example](#example)
-    - [How to use IAMs as authentication method](#how-to-use-iams-as-authentication-method)
```

Comment:
I know we already had this here, but shouldn't it be "IAM"?

More generally I think we might want to align this on other docs and say "Amazon OpenSearch Service" as a title.